src/Codeception/SuiteManager.php

```diff
@@ -178,28 +177,28 @@ protected function getActor()
             : $this->settings['class_name'];
     }
 
-    protected function checkEnvironmentExists(\Codeception\TestCase $test)
+    protected function checkEnvironmentExists(TestInterface $test)
     {
-        $envs = $test->getEnvironment();
+        $envs = $test->getMetadata()->getEnv();
         if (empty($envs)) {
             return;
         }
         if (!isset($this->settings['env'])) {
-            Notification::warning("Environments are not configured", TestCase::getTestFullName($test));
+            Notification::warning("Environments are not configured", Descriptor::getTestFullName($test));
             return;
         }
         $availableEnvironments = array_keys($this->settings['env']);
-        $listedEnvironments = explode(',', implode(',', $test->getEnvironment()));
+        $listedEnvironments = explode(',', implode(',', $envs));
         foreach ($listedEnvironments as $env) {
             if (!in_array($env, $availableEnvironments)) {
-                Notification::warning("Environment $env was not configured but used in test", TestCase::getTestFullName($test));
+                Notification::warning("Environment $env was not configured but used in test", Descriptor::getTestFullName($test));
             }
         }
     }
 
-    protected function isExecutedInCurrentEnvironment(\Codeception\TestCase $test)
+    protected function isExecutedInCurrentEnvironment(TestInterface $test)
     {
-        $envs = $test->getEnvironment();
+        $envs = $test->getMetadata()->getEnv();
         if (empty($envs)) {
             return true;
         }
```

Comment:
Actually, no. We check that we environments in tests are valid but we don't set them. So `setEnv` is not called